openedx/stanford/lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -0,0 +1,30 @@
+"""
+Custom override of SearchFilterGenerator without microsite org support
+using course tiles for discovery search
+"""
+from search.filter_generator import SearchFilterGenerator
+
+from branding_stanford.models import TileConfiguration
+from lms.lib.courseware_search.lms_filter_generator import LmsSearchFilterGenerator
+
+
+class TileSearchFilterGenerator(LmsSearchFilterGenerator):
+    """ SearchFilterGenerator for LMS Search """
+
+    def field_dictionary(self, **kwargs):
+        """ Return field filter dictionary for search """
+        # Adds course filter if provided, for in course search
+        field_dictionary = SearchFilterGenerator.field_dictionary(self, **kwargs)
```

Comment:
You mentioned that this is lifted from the LMS default,
but I see that we now omit the call to `super`.
- Was this an explicit omission?
- What is the impact/result of this change?

I'm wondering if we shouldn't do the same thing they do in the default;
start by grabbing the result of the `super` call,
then basically post-process it after w/ our customization.

Generally speaking, if a parent class also already calls up to the grandparent's method,
you generally want to also have the child call up to the parents method,
as the GP call indicates that the children depend on some logic having been executed "upstream".

Reply:
Yes I should clarify I only subclasses the default `LmsSearchFilterGenerator` so I wouldn't have to duplicate the enrollment code https://github.com/Stanford-Online/edx-platform/blob/master/lms/lib/courseware_search/lms_filter_generator.py#L22
but since we don't care about the org filter logic that it has I called the base class `field_dictionary` instead.